tests/Common/TriggersTest.php

```diff
@@ -393,28 +369,65 @@ public function testUpdateTriggerWithDifferentNonMasterToken()
         $anotherClientWithAnotherToken = $this->getClient([
             'url' => STORAGE_API_URL,
             'token' => $anotherToken['token'],
         ]);
 
-        try {
-            $anotherClientWithAnotherToken->updateTrigger((int) $trigger['id'], $updateData);
-            self::fail('should fail');
-        } catch (Exception $e) {
-            self::assertEquals('Insufficient privilege. This token cannot manipulate with this trigger', $e->getMessage());
-        }
+        $updatedTrigger = $anotherClientWithAnotherToken->updateTrigger((int) $trigger['id'], $updateData);
 
-        // master token can do anything
-        $updatedTrigger = $this->_client->updateTrigger((int) $trigger['id'], $updateData);
         self::assertEquals('keboola.ex-1', $updatedTrigger['component']);
         self::assertEquals(111, $updatedTrigger['configurationId']);
         self::assertEquals(20, $updatedTrigger['coolDownPeriodMinutes']);
         self::assertEquals($brandNewToken['id'], $updatedTrigger['runWithTokenId']);
         self::assertEquals([['tableId' => 'in.c-API-tests.watched-1']], $updatedTrigger['tables']);
     }
 
+    /**
+     * @return void
+     */
+    public function testUpdateTriggerWithNonMasterWithoutPrivilegeToken()
+    {
+        $table1 = $this->createTableWithRandomData("watched-1");
+
+        $optionsForTokenRunWith = (new TokenCreateOptions())
+            ->addBucketPermission($this->getTestBucketId(), TokenAbstractOptions::BUCKET_PERMISSION_READ);
+        $optionsForMainToken = (new TokenCreateOptions())->setCanManageBuckets(true);
+
+        $newNonAdminToken = $this->tokens->createToken($optionsForMainToken);
+        $clientWithoutAdminToken = $this->getClient(['url' => STORAGE_API_URL, 'token' => $newNonAdminToken['token']]);
+        $tokenRunWith = $this->tokens->createToken($optionsForTokenRunWith);
+
+
+        $trigger = $clientWithoutAdminToken->createTrigger([
+            'component' => 'orchestrator',
+            'configurationId' => 123,
+            'coolDownPeriodMinutes' => 10,
+            'runWithTokenId' => $tokenRunWith['id'],
+            'tableIds' => [$table1],
+        ]);
+
+        $optionsForTokenRunWith = (new TokenCreateOptions());
+        $newNonAdminToken = $this->tokens->createToken($optionsForTokenRunWith);
+        $anotherClientWithoutPrivilage = $this->getClient(['url' => STORAGE_API_URL, 'token' => $newNonAdminToken['token']]);
+
+        $updateData = [
+            'component' => 'orchestrator',
+            'configurationId' => 111,
+            'coolDownPeriodMinutes' => 20,
+            'runWithTokenId' => $tokenRunWith['id'],
+            'tableIds' => [$table1],
+        ];
+
+        try {
+            $anotherClientWithoutPrivilage->updateTrigger((int) $trigger['id'], $updateData);
+            self::fail('should fail');
+        } catch (Exception $e) {
+            self::assertEquals('Your token does not have sufficient privilege.', $e->getMessage());
+        }
```

Comment:
Proč to nestačí? Má přístup k té komponentě kterou ten trigger firuje, ne?

Reply:
ten token nemal ani jedno opravnenie preto to neslo, este som tie testy trochu prekopal.